apps/storefront/src/components/outSideComponents/B3CompanyHierarchyExternalButton.tsx

```diff
@@ -0,0 +1,199 @@
+
+  const isAddBottom = bottomHeightPage.some((item: string) => hash.includes(item));
+
+  const {
+    state: { switchAccountButton },
+  } = useContext(CustomStyleContext);
+
+  const defaultLocation: SnackbarOrigin = {
+    vertical: 'bottom',
+    horizontal: 'left',
+  };
+
+  const {
+    color = '',
+    customCss = '',
+    location = 'bottomLeft',
+    horizontalPadding = '',
+    verticalPadding = '',
+  } = switchAccountButton;
+
+  const cssInfo = splitCustomCssValue(customCss);
+  const {
+    cssValue,
+    mediaBlocks,
+  }: {
+    cssValue: string;
+    mediaBlocks: string[];
+  } = cssInfo;
+  const MUIMediaStyle = setMUIMediaStyle(mediaBlocks);
+
+  const baseStyles: SxProps = {
+    backgroundColor: color || '#ED6C02',
+    color: getContrastColor(color || '#FFFFFF'),
+    bottom: 0,
+    ...getStyles(cssValue),
+  };
+
+  const desktopStyles: SxProps = isAddBottom ? { bottom: '90px !important' } : {};
+  const buyerPortalStyles: SxProps = {
+    bottom: '24px',
+    left: '24px',
+    right: 'auto',
+    top: 'unset',
+  };
+  const mobileOpenStyles: SxProps = {
+    width: '100%',
+    bottom: 0,
+    left: 0,
+  };
+
+  let sx: SxProps = { ...baseStyles };
+
+  if (isMobile) {
+    if (isOpen) {
+      sx = { ...sx, ...mobileOpenStyles };
+    } else {
+      sx = { ...sx, ...getPosition(horizontalPadding, verticalPadding, location) };
+    }
+  } else if (isOpen) {
+    sx = { ...sx, ...buyerPortalStyles, ...desktopStyles };
+  } else {
+    sx = { ...sx, ...getPosition(horizontalPadding, verticalPadding, location), ...desktopStyles };
+  }
+
+  const companyName: string = useMemo(() => {
+    if (!selectCompanyHierarchyId) {
+      return '';
+    }
+
+    return (
+      companyHierarchyList.find((company) => company.companyId === +selectCompanyHierarchyId)
+        ?.companyName || ''
+    );
+  }, [selectCompanyHierarchyId, companyHierarchyList]);
+
+  const handleHierarchyExternalBtnClick = () => {
+    const { companyHierarchy } = pagesSubsidiariesPermission;
+
+    if (companyHierarchy) {
+      const { COMPANY_HIERARCHY } = PATH_ROUTES;
+
+      setOpenPage({
+        isOpen: true,
+        openUrl: COMPANY_HIERARCHY,
+      });
+
+      return;
+    }
+
+    const key = Object.keys(pagesSubsidiariesPermission).find((key) => {
+      return !!pagesSubsidiariesPermission[key as keyof PagesSubsidiariesPermissionProps];
+    });
+
+    const route = PAGES_SUBSIDIARIES_PERMISSION_KEYS.find((item) => item.key === key);
+
+    if (route && !isOpen) {
+      setOpenPage({
+        isOpen: true,
+        openUrl: route.path,
+      });
+    }
+    dispatch(setOpenCompanyHierarchyDropDown(true));
+  };
+
+  return (
+    <>
+      {!!companyName && (
+        <Snackbar
+          sx={{
+            zIndex: Z_INDEX.NOTIFICATION,
+            height: '52px',
+            borderRadius: '4px',
+            fontSize: '16px',
+            fontFamily: '"Roboto","Helvetica","Arial",sans-serif',
```

Comment:
This seems like it should be fixed at a layout level component - why is this styling logic being set at the snackbar component? and being rendered by this component.

Reply:
This component is independent of layout.
It will also appear when the buyer portal is closed.
It does not inherit the font style of the iframe, so a separate definition is made.